engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/exception/ExceptionLogger.java

```diff
@@ -20,36 +20,70 @@
 import java.io.PrintWriter;
 import java.io.StringWriter;
 import java.io.Writer;
-import java.util.logging.Level;
+import java.sql.SQLException;
 
+import org.camunda.bpm.engine.ProcessEnginePersistenceException;
 import org.camunda.commons.logging.BaseLogger;
 
+import static org.camunda.bpm.engine.impl.util.ExceptionUtil.PERSISTENCE_CONNECTION_ERROR_CLASS;
+
 public class ExceptionLogger extends BaseLogger {
 
   public static final String PROJECT_CODE = "ENGINE-REST";
   public static final String REST_API = "org.camunda.bpm.engine.rest.exception";
+
   public static final ExceptionLogger REST_LOGGER = BaseLogger.createLogger(ExceptionLogger.class,
-                                                                               PROJECT_CODE,
-                                                                               REST_API,
-                                                                               "HTTP");
+      PROJECT_CODE,
+      REST_API,
+      "HTTP"
+  );
 
   public void log(Throwable throwable) {
     Response.Status status = ExceptionHandlerHelper.getInstance().getStatus(throwable);
     int statusCode = status.getStatusCode();
 
-    if ( statusCode >= 500) {
-      logWarn(String.valueOf(statusCode), getStackTrace(throwable));
-    } else {
+    if (statusCode < 500) {
       logDebug(String.valueOf(statusCode), getStackTrace(throwable));
+      return;
     }
+
+    logInternalServerAndOtherStatusCodes(throwable, statusCode);
   }
 
-  protected String getStackTrace(Throwable aThrowable) {
+  protected void logInternalServerAndOtherStatusCodes(Throwable throwable, int statusCode) {
+
+    if (isPersistenceConnectionError(throwable)) {
+      logError(String.valueOf(statusCode), getStackTrace(throwable));
+      return;
+    }
 
-    final Writer      result      = new StringWriter();
+    logWarn(String.valueOf(statusCode), getStackTrace(throwable));
+  }
+
+  protected String getStackTrace(Throwable throwable) {
+    final Writer result = new StringWriter();
     final PrintWriter printWriter = new PrintWriter(result);
-    aThrowable.printStackTrace(printWriter);
+    throwable.printStackTrace(printWriter);
 
     return result.toString();
   }
```

Comment:
🔧 We already have `ExceptionUtil#getExceptionStacktrace`, ideally we'd reuse that here if possible.

Reply:
Done 👍